pkg/controller/pcideviceclaim/pcideviceclaim_controller.go

```diff
@@ -100,118 +134,204 @@ func unbindPCIDeviceFromVfioPCIDriver(addr string) error {
 	return nil
 }
 
-func (h Handler) reconcilePCIDeviceClaims(nodename string) error {
-	// Get all PCI Device Claims
-	var pdcNames map[string]string = make(map[string]string)
-	pdcs, err := h.pdcClient.List(metav1.ListOptions{})
+func addHostDeviceToKubeVirtAllowList(pd *v1beta1.PCIDevice) error {
+	logrus.Infof("Adding %s to KubeVirt Allow-List", pd.Name)
+	clientConfig := kubecli.DefaultClientConfig(&pflag.FlagSet{})
+	virtClient, err := kubecli.GetKubevirtClientFromClientConfig(clientConfig)
 	if err != nil {
-		return err
+		msg := fmt.Sprintf("cannot obtain KubeVirt client: %v\n", err)
+		return errors.New(msg)
+	}
+	ns := "harvester-system"
+	cr := "kubevirt"
+	kv, err := virtClient.KubeVirt(ns).Get(cr, &v1.GetOptions{})
+	if err != nil {
+		msg := fmt.Sprintf("cannot obtain KubeVirt CR: %v\n", err)
+		return errors.New(msg)
+	}
+	kvCopy := kv.DeepCopy()
+	if kv.Spec.Configuration.PermittedHostDevices == nil {
+		kvCopy.Spec.Configuration.PermittedHostDevices = &kubevirtv1.PermittedHostDevices{
+			PciHostDevices: []kubevirtv1.PciHostDevice{},
+		}
+	}
+	permittedPCIDevices := kv.Spec.Configuration.PermittedHostDevices.PciHostDevices
+	resourceName := pd.Status.ResourceName
+	// check if device is currently permitted
+	var devPermitted bool = false
+	for _, permittedPCIDev := range permittedPCIDevices {
+		if permittedPCIDev.ResourceName == resourceName {
+			devPermitted = true
+		}
+	}
+	vendorId := pd.Status.DeviceId
```

Comment:
`vendorId` and `deviceId` are both from `pd.Status.DeviceId`, ?

Reply:
nope, that's an error, fixed, will push up new commit soon